### ISSUE TITLE
API requests in `web/server/middleware/auth.js` will break with Authorization

### ISSUE THREAD

**Issue by makinde** [yekrojam/web#1]
Was thinking the other day and just looked. Your code makes a few API calls in `web/server/middleware/auth.js`. Those requests don't have a valid user yet. As current configured in the authorization diff, those calls will return nothing since it'll treat those calls as from a logged out user.

I'm not sure what the first call that looks for the org is for.

The second call that is for looking up the user by email address is one that I think the API can account for. My suggestion fro that case is to put the email address in the token that is sent to the API. The API can then do an extra query to convert that email address back into an ID so the rest of the code can work.

What do you think? I'll try to submit a diff for the change on the API side.

**Comment by ericgio** [comment]
The first call gets the org info to display in the app and **may** not have a valid user, but often will. The second call definitely **will not** have a valid user, since that's the query I'm using to find the user to log them in to begin with.

> put the email address in the token that is sent to the API

That feels like a workaround/hack to me. Even just querying the `/user` endpoint the way I'm currently doing is a workaround. There should be a first-class `/login` endpoint that expects a logged-out user, queries the user by email, and returns a user object.

**Comment by ericgio** [comment]
Yes, sorry that should have been "_I think_ there should be".

> what data should be exposed and to whom

Good question. Definitely makes sense to discuss in person, since I anticipate this will get pretty in-depth. In our last discussion I brought up the idea of an org or client context in addition to the existing user context. I think something like that would potentially address a lot of questions around permissions.

> If logged out people can query the database via email address

I'm a little confused about the mechanism here; how would someone do that? The way I have the request set up, the user has to authenticate via Auth0 first. Auth0 then sends user data (including email) to the client, which queries the API via email address. So someone can't just enter random email addresses (unless they also know the pw) because they'd fail Auth0 authentication. So in that sense, they _are_ logged in via Auth0, the client just doesn't have their id yet to send to the API. It sounds like what you're saying is that the API should accept either a user id or email as a valid claim when making a request?

**Comment by makinde** [comment]
Yeah, we can dive into the first one in person.

The second one might be an "in person" as well...partially because I'm very hazy on how easy it is for someone to get a valid `authToken` that contains `{}`. If landing on a logged out page will get a valid and empty `authToken` sent to browser, then I'm treating that as if anyone stranger can get a valid, empty authToken. I'd assumed that was the way it works, but a quick search of the codebase doesn't support that, so I'm not sure. If it is the case, then a random person could land on a logged out page, get their empty auth token, and start hitting the API with it to try to get `User` objects by querying email addresses. That's the "attack" I'm worried about. That said, if it's not easy to get a valid authToken that's empty, that "attack" is not possible.

**Comment by ericgio** [comment]
Ah I see. Yeah, it is currently possible to get a valid token while logged out.

**Comment by ericgio** [comment]
> it seems pretty reasonable to allow them access to all the fields currently in the Org model

Agreed, though I think in the SaaS case a logged out viewer shouldn't be able to get a list of orgs in the system. In the single-org case, they'd be able to see the info for just that org.
